### PR TITLE
feat: Text rendering, icon fonts, and image support (Phase 5)

### PR DESCRIPTION
**Pull request by jtp184**
## Summary

Phase 5 of the groundwork roadmap: adds text rendering via FreeType C bindings, icon font support with 1688 bundled Lucide glyphs, and image loading/transformation via optional ruby-vips. This enables rendering rich content (text, icons, images) onto Canvas surfaces for e-paper display output.

## Changes

- **`ext/chroma_wave/freetype.c` / `freetype.h`** — New C extension wrapping FreeType 2 for glyph rendering and measurement. Uses `TypedData_Wrap_Struct`, cached symbol IDs, pitch-aware bitmap copy, and safe shutdown ordering.
- **`ext/chroma_wave/extconf.rb`** — Optional FreeType 2 detection via `pkg_config` / header+lib fallback. Compiles with `-DNO_FREETYPE` when unavailable.
- **`lib/chroma_wave/font.rb`** — Ruby `Font` class with path/name resolution, thread-safe font discovery cache, `#measure` (multi-line aware), `#each_glyph`, and `#line_height`/`#ascent`/`#descent`.
- **`lib/chroma_wave/text_metrics.rb`** — Immutable `TextMetrics` value type via `Data.define`.
- **`lib/chroma_wave/drawing/text.rb`** — `Drawing::Text` mixin for `Surface` with `#draw_text`, word wrapping, alignment (`:left`/`:center`/`:right`), and source-over alpha compositing with integer fast path for opaque backgrounds.
- **`lib/chroma_wave/icon_font.rb`** — `IconFont < Font` with symbol-to-codepoint registry, `#draw`, `#measure_icon`, `#icon_names`.
- **`lib/chroma_wave/icon_font/lucide_glyphs.rb`** — Auto-generated 1688-glyph Lucide codepoint map.
- **`lib/chroma_wave/image.rb`** — `Image` class with optional ruby-vips dependency. `.load`/`.from_buffer`, `#resize`, `#crop`, `#draw_onto`, `#to_canvas`, full RGBA normalization (1/2/3/4-band + CMYK).
- **`data/fonts/`** — Bundled DejaVu Sans and Lucide icon font with licenses.
- **`lib/tasks/icons.rake`** — Rake task to regenerate Lucide glyph map from JSON metadata.
- **`.github/workflows/main.yml`** — CI installs `libfreetype-dev` and `libvips-dev` for all relevant jobs.
- **`spec/`** — 90+ new specs: Font, IconFont, TextMetrics, Drawing::Text, Image, content pipeline integration, and shared pixel helpers.

## Type of Change

- [x] `feat:` New feature (text rendering, icon fonts, image support)
- [x] `chore:` CI updates, bundled font assets, rake task
- [x] `fix:` Alpha blending correctness, GC pressure reporting, edge case handling

## Testing

- 849 specs, 0 failures
- 94.64% line coverage, 84.93% branch coverage
- Rubocop clean (0 offenses)
- Unit tests for Font (loading, discovery, measurement, edge cases), IconFont (draw, measure, glyph map), TextMetrics (immutability, equality), Drawing::Text (rendering, wrapping, alignment, anti-aliasing, newlines), Image (load, resize, crop, draw_onto, RGBA normalization, dependency error)
- Integration specs exercise the full Font → Canvas → Renderer → Framebuffer → Display pipeline
- FreeType-absent path tested via stubbed `respond_to?`
- Vips-absent path tested via `hide_const` + stubbed `require`

## Notes for Reviewers

- FreeType is an optional compile-time dependency — the gem builds and runs without it, raising `DependencyError` when text features are used.
- ruby-vips is an optional runtime dependency — `Image` lazily requires it and raises `DependencyError` with an install hint if missing.
- The FreeType library singleton is lazily initialized and cleaned up via `rb_set_end_proc`. A comment documents the thread-safety tradeoff (acceptable for e-paper setup-time usage).
- Alpha compositing uses an integer-only fast path when the background is opaque (the common case for e-paper canvases).
- The Lucide glyph map is auto-generated; the rake task and source JSON are included for reproducibility.

### 🖋️📐🎨🖼️✨

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -13,7 +13,10 @@
 
 require 'chroma_wave'
 
+Dir[File.join(__dir__, 'support', '**', '*.rb')].each { |f| require f }
+
```
[the remark this answers is not in the document]
Comment (jtp184) [reply spec/spec_helper.rb]: Not applicable — Ruby 3.0+ guarantees `Dir[]` returns sorted results. Adding `.sort` triggers `Lint/RedundantDirGlobSort`.
